lib/spack/env/cc

```diff
@@ -175,15 +175,26 @@ while [ -n "$1" ]; do
             ;;
         -Wl,*)
             arg="${1#-Wl,}"
-            if [ -z "$arg" ]; then shift; arg="$1"; fi
-            if [[ "$arg" = -rpath=* ]]; then
-                rpaths+=("${arg#-rpath=}")
-            elif [[ "$arg" = -rpath ]]; then
+            # TODO: Handle multiple -Wl, continuations of -Wl,-rpath
+            if [[ $arg == '-rpath='* ]]; then
```

Comment:
@eschnett: Does this handle the case when a directory name has escaped spaces in it?  I believe the original case did because it quoted `"$arg"`.  This version leaves `"$arg"` bare fro some reason.  Why is that?  Also why does `'-rpath='*` need to be quoted... wasn't the original construction simpler?